docs/spec_draft.md

```diff
@@ -1243,6 +1244,78 @@ Numeric precision is implementation-defined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.slice
+
+### Semantics
+
+Extracts a sub-tensor from the `operand` using the dimension indices from the
+operands.
+
+More formally, `result[i0, ..., ik, ..., iR-1] = operand[j0+s0, ..., jk+sk, ..., jR-1+sR-1]`
+for all `sk` in `strides` for all `jk` in `start_indices[k]` $\leq$ `jk` $\lt$
+`limit_indices[k]` where 0 $\leq$ `k` $\lt$ rank(`operand`).
+
+Shape is calculated by `dim(result, i)` = $\lceil$`(limit_indices[i]-start_indices[i])/stride[i]`$\rceil$
+for `i`th dimension size of `operand`.
+
+### Operands
+
+| Name            | Type                                                                                                                                    |
+|-----------------|-----------------------------------------------------------------------------------------------------------------------------------------|
+| `operand`       | tensor of any supported types                                                                                                           |
+| `start_indices` | 1-dimensional array of integers containing the starting indices of the slice for each dimension                                         |
+| `limit_indices` | 1-dimensional array of integers containing the ending indices (exclusive) for the slice for each dimension                              |
+| `strides`       | 1-dimensional array of integers that decides the input stride of the slice. The slice picks every `strides[d]` element in `d` dimension |
+
+### Results
+
+| Name     | Type                          |
+|----------|-------------------------------|
+| `result` | tensor of any supported types |
+
+### Constraints
+
+  * (C1) `operand` and `result` have the same element type.
+  * (C2) `start_indices`, `limit_indices`, and `strides` have the same type.
```

Comment:
C3 can be derived from C2

Reply:
What about the ```... = rank(`operand`)``` part? I wasn't sure how to merge (C2) and (C3) into one without making it seem to have multiple constraints in one line.

Reply:
What I meant was:  `start_indices`, `limit_indices`, and `strides` have the same type => have the same shape which for a 1-D tensor is its size.

Reply:
This constraint was more so meant to make sure that the rank of the `operand` is the same size as the shape/size of `start_indices`, etc. as opposed to making sure the type/shape of the three were the same. This constraint is needed since we could have arbitrarily large number of indices when the rank of `operand` is small. Maybe:

`start_indices`, `limit_indices`, and `strides` have the same type and their size is equal to rank of `operand`.

Reply:
Ah! I see you point. I missed the part $==$ rank(`operand`).

The suggestion makes sense.

Reply:
I think it's the other way around - C2 can be derived from C3 :)